source/atomic_ops.cpp

```diff
@@ -63,7 +62,56 @@ bool atomic_cas<uint32_t>(uint32_t *ptr, uint32_t *expectedCurrentValue, uint32_
 
     return !__STREXW(desiredValue, ptr);
 }
-#endif /* #if (__CORTEX_M >= 0x03) */
+
+uint32_t atomic_incr(uint32_t *valuePtr, uint32_t delta)
+{
+    uint32_t newValue;
+    do {
+        newValue = __LDREXW(valuePtr) + delta;
```

Comment:
LDREX and such are better encapsulated into a single abstraction: atomic_cas. Other atomic ops should re-use atomic_cas.
I disagree with this approach of duplicating the definitions of atomic_incr; a templatized definition is more suitable.
I also feel that adding the missing declarations for the template specializations (for atomic cas) is the simplest way to solve this whole issue.